eo-maven-plugin/src/test/java/org/eolang/maven/ProbeMojoTest.java

```diff
@@ -0,0 +1,191 @@
+            new LinkedList<>(new MnJson(foreign).read()).getFirst().get("hash"),
+            Matchers.equalTo("mmmmmmm")
+        );
+    }
+
+    @Test
+    void findsProbesViaOfflineHash(@TempDir final Path temp) throws IOException {
+        final Input src = new InputOf(
+            new TextOf(
+                new ResourceOf("org/eolang/maven/simple-io.eo")
+            )
+        );
+        this.saveProgram(temp, src);
+        this.execUntilProbeMojo(temp);
+        final File target = temp.resolve("target").toFile();
+        final File foreign = temp.resolve(ProbeMojoTest.FOREIGN).toFile();
+        new Moja<>(ProbeMojo.class)
+            .with("targetDir", target)
+            .with("foreign", foreign)
+            .with("foreignFormat", "json")
+            .with("objectionary", new OyFake())
+            .with("tag", "1.0.0")
+            .with("offlineHash", "*.*.*:abcdefg")
+            .execute();
+        MatcherAssert.assertThat(
+            new LinkedList<>(new MnJson(foreign).read()).getFirst().get("hash"),
+            Matchers.equalTo("abcdefg")
+        );
+    }
+
+    @Test
+    @ExtendWith(OnlineCondition.class)
+    void findsProbesInOyRemote(@TempDir final Path temp) throws IOException {
+        final Input src = new InputOf(
+            new TextOf(
+                new ResourceOf("org/eolang/maven/simple-io.eo")
+            )
+        );
+        this.saveProgram(temp, src);
+        this.execUntilProbeMojo(temp);
+        final File target = temp.resolve("target").toFile();
+        final File foreign = temp.resolve(ProbeMojoTest.FOREIGN).toFile();
+        final Objectionary obj = new OyRemote(new ChCached(new ChRemote("0.28.10")));
+        new Moja<>(ProbeMojo.class)
+            .with("targetDir", target)
+            .with("foreign", foreign)
+            .with("foreignFormat", "json")
+            .with("tag", "0.28.10")
+            .with("objectionary", obj)
+            .execute();
+        MatcherAssert.assertThat(
+            new LinkedList<>(new MnJson(foreign).read()).getFirst().get("probed"),
+            Matchers.equalTo("2")
+        );
+    }
+
+    private void execUntilProbeMojo(final Path temp) {
+        final File target = temp.resolve("target").toFile();
+        final File foreign = temp.resolve(ProbeMojoTest.FOREIGN).toFile();
+        new Moja<>(ParseMojo.class)
+            .with("targetDir", target)
+            .with("foreign", foreign)
+            .execute();
+        new Moja<>(OptimizeMojo.class)
+            .with("targetDir", target)
+            .with("foreign", foreign)
+            .execute();
+        new Moja<>(DiscoverMojo.class)
+            .with("targetDir", target)
+            .with("foreign", foreign)
+            .execute();
+    }
+
+    private void saveProgram(final Path temp, final Input code) throws IOException {
+        final Path program = temp.resolve("program.eo");
+        new Home(temp).save(code, temp.relativize(program));
```

Comment:
@MikhailLipanin It seems, we don't need to `relativize` program path here, since `program` is already an absolute. What do you think?

Reply:
fixed